### DISCUSSION TITLE
Feature: DatePicker

### DISCUSSION THREAD

**Discussion by shafkathullah** [tailwindlabs/headlessui#289] [Ideas]
A headless DatePicker will be great!

**Comment by changesbyjames** [comment]
I've been using headlessui since pre-release and I love it's flexibility so I've been seeking out other libraries that attempt to solve a similar thing. I've been using this [headless calendar/date picker hook](https://github.com/its-danny/use-lilius) in a recent project. It's not super used and has some gnarly edge cases but it could be a good starting point for the design?

**Comment by brielov** [comment]
Any news on this?

**Comment by itstimwhite** [comment]
New calenders in Tailwind UI look great, would love to see a headless Ui component.

**Comment by shafkathullah** [comment]
<img width="1615" alt="Screenshot 2022-03-29 at 11 09 55 AM" src="https://user-images.githubusercontent.com/15883435/160541034-da03d60f-ec4f-4274-a928-5d9702cd4bc9.png">
officially It's not in plan at least this time

**Comment by rubber-duck-software** [comment]
A custom date input component is a very common need in a project and so several component libraries include this in their standard set of components. See

* https://mui.com/x/react-date-pickers/date-field/
* https://ant.design/components/date-picker
* https://buefy.org/documentation/datepicker

Now, if HTML includes a `<input type="date" />` why do component libraries feel the need to reinvent the wheel? Simply put, the HTML standard is not flexible enough for lots of applications. Much like `datalist` doesn't cut it for many Combobox scenarios and `select` doesn't cut it for many Listbox scenarios, `<input type="date" />` doesn't cut it for many DateInput scenarios.

Further, a good custom DateInput has a lot of logic to make it behave just right. This makes DateInput an ideal candidate for HeadlessUI treatment.

I propose a new HeadlessUI component called DateInput. It would behave similar to the Native date input in Safari and Chrome, in that, as the types, the cursor would be selecting different parts of the input to guide the user through typing a date.

I am already working on a Headless implementation of this and would be happy to create an initial PR if this idea is accepted

**Reply by aishik-pyne** [reply]
Hi @rubber-duck-software 

This would be a great feature to have and thanks for taking it up!
I'm in a need for this for myself as well, may I ask is this implemented now?

**Reply by rubber-duck-software** [reply]
As far as I'm aware this feature request is still awaiting a "go-ahead" from any of the headless ui owners. 

I still believe this would be a worthwhile addition, but a date picker is a lot to maintain, so the owners might not be interested. 

In the meantime, I'd encourage you to check out https://react-spectrum.adobe.com/react-aria/examples/datepicker.html#datepicker

**Comment by aliakbarazizi** [comment]
Hey, I created a new library for datepicker
check it out and let me know if you have any issues
https://github.com/aliakbarazizi/headless-datepicker

it support any kind of picker or calendar
you can see all in storybook demo pages

**Comment by Askarov5** [comment]
Any updates on this?

**Reply by shafkathullah** [reply]
I now use react aria components, which has best support for date/calendar related stuff, highly customisable.

**Comment by adambirds** [comment]
It sucks that this hasn't been implemented yet. Its nearly impossible to style the standard date input.

**Reply by shafkathullah** [reply]
i use https://originui.com/ these days, they are just 💯

**Reply by exSnake** [reply]
> i use https://originui.com/ these days, they are just 💯

you switch component every year? 😂

**Reply by shafkathullah** [reply]
haha, i mean we have a goto lib every moment right, and these days its baseUI,
